cpp/arcticdb/util/hash.hpp

```diff
@@ -10,9 +10,17 @@
 #include <cstdint>
 #include <string_view>
 
+#ifdef ARCTICDB_USING_STATIC_XXHASH
 #define XXH_STATIC_LINKING_ONLY
+#else
+#define XXH_INLINE_ALL
+#endif 
 #include <xxhash.h>
+#ifdef ARCTICDB_USING_STATIC_XXHASH
 #undef XXH_STATIC_LINKING_ONLY
+#else
+#undef XXH_INLINE_ALL
+#endif 
```

Comment:
@DerThorsten: could you add a comment to explain the motivation for this change here? :slightly_smiling_face:

Reply:
it makes it compile =)

Reply:
Well obviously. :upside_down_face: 

I was more thinking of this lines of:

```suggestion
// Adapt macro inlining depending on xxhash's linkage for the build.
#ifdef ARCTICDB_USING_STATIC_XXHASH
#define XXH_STATIC_LINKING_ONLY
#else
#define XXH_INLINE_ALL
#endif 
#include <xxhash.h>
#ifdef ARCTICDB_USING_STATIC_XXHASH
#undef XXH_STATIC_LINKING_ONLY
#else
#undef XXH_INLINE_ALL
#endif 
```

Do you think it is useful for maintenance?